src/app/core/services/navigation/group-nav-tree.service.ts

```diff
@@ -23,14 +23,24 @@ export class GroupNavTreeService extends NavTreeService<GroupInfo> {
     super(currentContent);
   }
 
+  childrenNavigation(): OperatorFunction<GroupInfo|undefined,GroupNavigationData|undefined> {
+    return pipe(
+      distinctUntilChanged((g1, g2) => g1?.route.id === g2?.route.id),
+      switchMap(group => {
+        if (!group || group.route.isUser) return of(undefined);
+        return concat(of(undefined), this.groupNavigationService.getGroupNavigation(group.route.id));
+      })
+    );
+  }
+
   isOfContentType(content: ContentInfo|null): content is GroupInfo {
     return isGroupInfo(content);
   }
 
-  addDetailsToTreeElement(contentInfo: GroupInfo, treeElement: NavTreeElement): NavTreeElement {
+  addDetailsToTreeElement(treeElement: NavTreeElement, contentInfo: GroupInfo, children?: GroupNavigationData): NavTreeElement {
```

Comment:
Then it may be hard to understand why we do not try to get nav data when the item is not a chapter/skill... We could call that "childrenNavData" if you prefer, but that becomes heavier.